tests/unit/test_io_hdf5_h5tools.py

```diff
@@ -1577,7 +1577,7 @@ def test_link(self):
 
             self.assertTrue(self.foo2.my_data.valid)  # test valid
             self.assertEqual(len(self.foo2.my_data), 5)  # test len
-            self.assertEqual(self.foo2.my_data.shape, (5,))  # test getattr with shape
+            self.assertEqual(self.foo2.my_data.size, (5,))  # test getattr with size
```

Comment:
I am surprised this works. When I put a breakpoint here, I can see that `self.foo2.my_data.size` is 5, and `self.assertEqual(self.foo2.my_data.size, (5,))` raises an AssertionError. So why does this pass???

Reply:
```
            self.assertEqual(self.foo2.my_data.size, (5,))  # test getattr with size
            self.assertEqual(self.foo2.my_data.size, 5)  # test getattr with size
>           self.assertEqual(5, (5,))  # test getattr with size
E           AssertionError: 5 != (5,)
```
That's weird.

Reply:
```python
type(self.foo2.my_data.size)  # np.int64
self.assertEqual(np.int64(5), (5,))  # True
```

Reply:
This is probably why. A little gotcha when comparing numpy ints with other types
```python
>>> 5 == (5, )
False
>>> np.int64(5) == (5, )
array([ True])
>>> bool(np.int64(5) == (5,))
True
```